cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/ckan/CKANBackendImpl.java

```diff
@@ -79,8 +81,12 @@ public void persist(String orgName, String pkgName, String resName, String recor
         throws CygnusBadConfiguration, CygnusRuntimeError, CygnusPersistenceError {
         LOGGER.debug("Going to lookup for the resource id, the cache may be updated during the process (orgName="
                 + orgName + ", pkgName=" + pkgName + ", resName=" + resName + ")");
-        String resId = resourceLookupOrCreate(orgName, pkgName, resName, createEnabled);
-        
+        String resId = "";
+        if (!createEnabled){
+            resId= resourceLookupOrCreateDynamicFields(orgName, pkgName, resName,records);
+        } else {
+        resId = resourceLookupOrCreate(orgName, pkgName, resName, createEnabled);
```

Comment:
Please check indentation (github is not showing this line correctly indented, maybe it's due to some tabs vs space issue)

Reply:
Fixed